recipe/recipe.yaml

```
@@ -10,11 +14,28 @@ source:
   sha256: 314bb28890c36a151d4cae87b4077f0900f7d306880104b0fbbcbd0a8ae5b9ed
 
 build:
-  number: 0
+  number: ${{ ( (build_num | int) + 100 * (mlevel | int) + (1000 if use_fast_math else 0) ) }}
+  python:
+    version_independent: true
   skip:
     - win
+    - not is_python_min
+    - microarch_level > 1 and not (unix and x86_64)
   script:
-    - export CMAKE_CONFIGURE_ARGS="${CMAKE_ARGS} -DBLAS_LIBRARIES=lapack;blas -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=$PREFIX -DCMAKE_INSTALL_LIBDIR=lib -DBUILD_SHARED_LIBS=ON -DBUILD_LIBCINT=OFF -DBUILD_LIBXC=OFF -DBUILD_XCFUN=OFF -LAH"
+    - if: use_fast_math
+      then:
+        # this contains "-ffast-math"
+        # libmvec's vector math was introduced in glibc 2.22,
+        - cp $SRC_DIR/pyscf/lib/cmake_user_inc_examples/cmake.user.inc-gnu $SRC_DIR/pyscf/lib/cmake.arch.inc
+        # need to link libm for -ffast-math
+        - echo "link_libraries(m)" >> $SRC_DIR/pyscf/lib/cmake.arch.inc
+    - if: (microarch_level == 4)
+      then:
+        # manually add the appropriate flags so we can cross compile for v4
+        - export CFLAGS="${CFLAGS} -march=x86-64-v4"
+        - export CXXFLAGS="${CXXFLAGS} -march=x86-64-v4"
+        - export CPPFLAGS="${CPPFLAGS} -march=x86-64-v4"
+    - export CMAKE_CONFIGURE_ARGS="${CMAKE_ARGS} -DBLAS_LIBRARIES=lapack;blas -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=$PREFIX -DCMAKE_INSTALL_LIBDIR=lib -DBUILD_SHARED_LIBS=ON -DBUILD_LIBCINT=OFF -DBUILD_LIBXC=OFF -DBUILD_XCFUN=OFF -DCMAKE_VERBOSE_MAKEFILE=ON -LAH"
```

These two options are already part of CMAKE_ARGS
```suggestion
    - export CMAKE_CONFIGURE_ARGS="${CMAKE_ARGS} -DBLAS_LIBRARIES=lapack;blas -DCMAKE_INSTALL_LIBDIR=lib -DBUILD_SHARED_LIBS=ON -DBUILD_LIBCINT=OFF -DBUILD_LIBXC=OFF -DBUILD_XCFUN=OFF -DCMAKE_VERBOSE_MAKEFILE=ON -LAH"
```